core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -434,7 +433,7 @@ class BrokerServer(
         maybeChangeStatus(STARTING, STARTED)
         fatal("Fatal error during broker startup. Prepare to shutdown", e)
         shutdown()
-        throw e
+        throw if (e.isInstanceOf[ExecutionException]) e.getCause else e
```

Comment:
This causes the KRaft broker to throw KafkaStorageException like the ZK broker rather than ExecutionException, which is what is thrown now.

Reply:
yes, this is a good change :)